addons/payment_mollie/models/payment_acquirer.py

```diff
@@ -0,0 +1,82 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import logging
+
+import requests
+from werkzeug import urls
+
+from odoo import _, api, fields, models, service
+from odoo.exceptions import ValidationError
+
+from odoo.addons.payment_mollie.const import SUPPORTED_CURRENCIES
+
+_logger = logging.getLogger(__name__)
+
+
+class PaymentAcquirer(models.Model):
+    _inherit = 'payment.acquirer'
+
+    provider = fields.Selection(
+        selection_add=[('mollie', 'Mollie')], ondelete={'mollie': 'set default'}
+    )
+    mollie_api_key = fields.Char(
+        string="API key",
+        help="The Test or Live API Key depending on the configuration of the acquirer",
+        required_if_provider="mollie", groups="base.group_system"
+    )
+
+    #=== BUSINESS METHODS ===#
+
+    @api.model
+    def _get_compatible_acquirers(self, *args, currency_id=None, **kwargs):
+        """ Override of payment to unlist Mollie acquirers for unsupported currencies. """
+        acquirers = super()._get_compatible_acquirers(*args, currency_id=currency_id, **kwargs)
+
+        currency = self.env['res.currency'].browse(currency_id).exists()
+        if currency:
+            acquirers = acquirers.filtered(
+                lambda a: a.provider != 'mollie' or currency.name in SUPPORTED_CURRENCIES
+            )
```

Comment:
```suggestion
        if currency and currency.name not in SUPPORTED_CURRENCIES:
            acquirers = acquirers.filtered(lambda a: a.provider != 'mollie')
```
Move the generic condition outside the filtered lambda function, no need to evaluate it for each acquirer in the recordset ;).